sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -1803,7 +1811,25 @@ private async Task StartPublishingAsync(CancellationToken cancellationToken)
 
                 if ((_partitions == null) || (_partitionHash == null))
                 {
-                    await UpdatePartitionInformation(cancellationToken).ConfigureAwait(false);
+                    // The retry policy for the buffered producer is, by default, generous to allow for long running background work to be done
+                    // effectively. For startup, a more conservative retry policy is needed to prevent the producer from hanging, so cancel
+                    // after one TryTimeout interval.
+
+                    var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+                    linkedCts.CancelAfter(_options.RetryOptions.TryTimeout);
+
+                    try
+                    {
+                        await UpdatePartitionInformation(linkedCts.Token).ConfigureAwait(false);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        throw new EventHubsException(true, EventHubName, "Querying the Event Hub metadata took longer than expected and failed.", EventHubsException.FailureReason.ServiceCommunicationProblem);
```

Comment:
I would make this a `ServiceTimeout` reason code and tweak the verbiage to something like "The buffered producer took too long to start."